frame/support/src/traits/tokens/nonfungible.rs

```diff
@@ -0,0 +1,190 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2019-2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Traits for dealing with a single non-fungible asset class.
+//!
+//! This assumes a single level namespace identified by `Inspect::InstanceId`, and could
+//! reasonably be implemented by pallets which wants to expose a single collection of NFT-like
+//! objects.
+//!
+//! For an NFT API which has dual-level namespacing, the traits in `nonfungibles` are better to
+//! use.
+
+use codec::{Encode, Decode};
+use sp_std::prelude::*;
+use sp_runtime::TokenError;
+use crate::dispatch::DispatchResult;
+use crate::traits::Get;
+use super::nonfungibles;
+
+/// Trait for providing an interface to a read-only NFT-like set of asset instances.
+pub trait Inspect<AccountId> {
+	/// Type for identifying an asset instance.
+	type InstanceId;
```

Comment:
This will make working with `<T as Inspect>::InstanceId` pretty difficult, no? Maybe we should consider adding a few trait bounds?
Some candidates:
```Rust
Encode, Decode, Eq, Ord, Clone
```